src/core/src/pass/stateful_to_stateless.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/pass/stateful_to_stateless.hpp"
+
+#include <regex>
+#include <string>
+
+#include "openvino/cc/pass/itt.hpp"
+#include "openvino/op/assign.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/read_value.hpp"
+#include "openvino/pass/manager.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace ov::op;
+
+namespace {
+
+std::shared_ptr<ov::Node> setName(std::shared_ptr<ov::Node> node, const std::string& name) {
+    // Set name for both node and output tensor (should be only one tensor, and any other names will be overriden by a
+    // given single name)
+    node->set_friendly_name(name);
+    OPENVINO_ASSERT(node->get_output_size() == 1);
+    node->get_output_tensor(0).set_names({name});
+    return node;
+}
+
+// Templated method that has the same effect as not templated `setName` but saves Op type for convenient calls chaining
+template <typename T>
+inline std::shared_ptr<T> setName(std::shared_ptr<T> node, const std::string& name) {
+    setName(std::dynamic_pointer_cast<ov::Node>(node), name);
+    return node;
+}
```

Comment:
@p-durandin, it should be a part of core utils, I needed to write it already for the second time. Please create a task for moving it to the common core utils with appropriate unification/generalization.